packages/target-tauri/src-tauri/src/temp_file.rs

```diff
@@ -163,23 +176,41 @@ pub(crate) async fn remove_temp_file(app: AppHandle, path: SafePathBuf) -> Resul
 }
 
 //copyBlobFileToInternalTmpDir
+/// It fails if `file_name` is not a plain name (e.g. includes parent dir);
+/// if `source_path` is not a file inside of one of accounts' blobs dir.
 #[tauri::command]
 pub(crate) async fn copy_blob_file_to_internal_tmp_dir(
     app: AppHandle,
     dc: State<'_, DeltaChatAppState>,
     file_name: &str,
-    source_path: PathBuf,
+    source_path: SafePathBuf,
 ) -> Result<String, Error> {
-    if !source_path.starts_with(&dc.accounts_dir)
-        || !source_path.components().any(|c| {
-            if let Some(c) = c.as_os_str().to_str() {
-                c.ends_with("-blobs")
-            } else {
-                false
-            }
-        })
+    let source_path = source_path.as_ref();
+
     {
-        return Err(Error::SourcePathOutsideOfBlobsDir);
+        let maybe_blobs_dir = source_path
+            .parent()
+            .ok_or(Error::SourcePathOutsideOfBlobsDir)?;
+        if maybe_blobs_dir
+            .file_name()
+            .ok_or(Error::SourcePathOutsideOfBlobsDir)?
+            != "dc.db-blobs"
```

Comment:
I choose the ends with intentional as I thought someone might possible have an db with a different name, though thinking about it that edge-case will probably not happen, so checking for exact name is also fine